code/modules/food_and_drinks/kitchen_machinery/griddle.dm

```diff
@@ -63,6 +63,37 @@
 
 
 /obj/machinery/griddle/attackby(obj/item/I, mob/user, params)
+	if(istype(I, /obj/item/storage/bag/tray))
+		var/datum/component/storage/tray_storage = I.GetComponent(/datum/component/storage)
+		if(!I.contents.len && griddled_objects.len > 0) //if the griddle has things on it and the tray is empty, put what you can into the tray
+			for(var/obj/item/grilled as anything in griddled_objects)
+				if(!griddled_objects.len)
+					return
+				if(I.contents.len >= tray_storage.max_items)
+					to_chat(user, span_notice("[I] can't fit more items!")) //tray is full
+					return
+				if(SEND_SIGNAL(I, COMSIG_TRY_STORAGE_INSERT, grilled, user))
+					ItemRemovedFromGrill(grilled)
+				else if(!SEND_SIGNAL(I, COMSIG_TRY_STORAGE_INSERT, grilled, user))
+					to_chat(user, span_notice("[I] can't hold [grilled]!")) //if an item cant enter the tray, dont touch it
+				else
+					return
```

Comment:
because the tray will be placed on the griddle if theres no returns, i cannot have any end part of the loops that dont return, because then the proc contines outside of the loop/if and then runs the rest of griddle code onto the tray: aka placing the tray onto the griddle